### PR TITLE
ELK stack bug fixes

### PR DESCRIPTION
**Pull request by rachelg4**

### CODE REVIEW

**Comment by rachelg4** [comment]
@stvstnfrd

**Comment by rachelg4** [comment]
Still need to add basic auth

File: playbooks/roles/logstash/templates/logstash.conf.j2 [hunk 1]
```diff
@@ -18,7 +18,7 @@ filter {
     }
   } else {
     multiline {
-      pattern => "(^(?![0-9]|[A-Z][a-z][a-z]\s|\[[0-9]|date|\{)|^([0-9][0-9]\s))"
+      pattern => "(^(?![0-9]|[A-Z][a-z][a-z]\s[0-9]|\[[0-9]|date|\{)|^([0-9][0-9]\s))"
```
Comment (stvstnfrd) [inline playbooks/roles/logstash/templates/logstash.conf.j2]: Add comment

File: stanford/logs-backend.yml [hunk 2]
```diff
@@ -4,6 +4,9 @@
   gather_facts: True
   vars:
     LOGSTASH_ELASTICSEARCH_HOST: localhost
+    # take in number of mega bytes, will scale it properly in elasticsearch/tasks 
+    ELASTICSEARCH_HEAP_SIZE_INT: "{{ ansible_memtotal_mb / 2 }}"
```
Comment (stvstnfrd) [inline stanford/logs-backend.yml]: I think you need to explicitly cast this as an integer.
When I installed to the sandbox, the value was `497.0m`

Elasticsearch crashed and refused to start until the decimal part was removed :(

File: playbooks/roles/filebeat/templates/filebeat.yml.j2 [hunk 3]
```diff
@@ -12,11 +12,14 @@ filebeat:
...
         - "{{ COMMON_LOG_DIR }}/cms/edx.log"
         - "{{ COMMON_LOG_DIR }}/lms/edx.log"
-        - "{{ COMMON_LOG_DIR }}/notifier/edx.log"
         - "{{ COMMON_LOG_DIR }}/xqueue/edx.log"
         - "{{ COMMON_LOG_DIR }}/certs/edx.log"
-      document_type: openedx_standard
+        - "{{ COMMON_LOG_DIR }}/notifier/edx.log"
+        - "{{ COMMON_LOG_DIR }}/analytics-api/edx.log"
+        - "{{ COMMON_LOG_DIR }}/insights/edx.log"
+      document_type: openedx_other
```
Comment (stvstnfrd) [inline playbooks/roles/filebeat/templates/filebeat.yml.j2]: Is there a more descriptive name we can use here?

File: playbooks/roles/logstash/templates/logstash.conf.j2 [hunk 4]
```diff
@@ -18,7 +18,8 @@ filter {
     }
   } else {
     multiline {
-      pattern => "(^(?![0-9]|[A-Z][a-z][a-z]\s|\[[0-9]|date|\{)|^([0-9][0-9]\s))"
+      # [A-Z][a-z][a-z]\s[0-9] looks ofr a month followed by a day of the form Feb 28
```
Comment (stvstnfrd) [inline playbooks/roles/logstash/templates/logstash.conf.j2]: `s/ofr/for/`

File: playbooks/roles/logstash/templates/logstash.conf.j2 [hunk 5]
```diff
@@ -48,25 +53,34 @@ filter {
...
-    } else if [type] == "supervisor_worker_error" {
+    } else if [type] == "supervisor_worker_cms_error" {
       grok {
         match => { "message" => "\[%{TIMESTAMP_ISO8601:syslog_timestamp}\: %{WORD:log_level}\/%{USERNAME:supervisor_function}\] %{GREEDYDATA:@message}" }
       }
+    } else if [type] == "supervisor_worker_lms_error" {
+      grok {
+        match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} %{WORD:log_level} %{POSINT:syslog_pid} \[%{USERNAME:supervisor_function}\] %{DATA:filename}\:%{INT:line_number} \- %{GREEDYDATA:@message}" }
+      }
+
```
Comment (stvstnfrd) [inline playbooks/roles/logstash/templates/logstash.conf.j2]: Nit: delete superfluous newline

**Comment by stvstnfrd** [comment]
One superfluous newline, otherwise +1 to squash, edit commit message, and merge

**Pull request merged**